### DISCUSSION TITLE
worker_concurrency on task level

### DISCUSSION THREAD

**Discussion by rmat345** [apache/airflow#30700] [Q&A]
We work with DAGs that include heavy load in parallel KubernetesPodOperators and lower load in BigQueryExecuteOperators on Google Cloud Composer with Airflow 2. To get the whole system in a stable and cost-efficient way, we need to specify a low worker_concurrency in Airflow. Otherwise the KubernetesPodOperators fail. However, for the BigQueryExecutorOperators the low worker_concurrency is a waste. Many more of them could run on one worker. Is there way to specify the worker_concurrency on the task level rather than as a global Airflow configuration?

**Comment by kobethuwis** [comment]
Airflow has a default task concurrency of 32, meaning that you can run at most 32 tasks in parallel at once (similar to worker concurrency for k8s or celery). Maybe you could have a look at task pooling in Airflow? We use it to i.e. limit `PostgresOperators `to on-prem database while still allowing lot's of `BaseOperators `through.

```
BaseOperatorPool with size 32
PostgresOperatorPool with size 8
```

Then you could define i.e. a standardized `PostgresOperator`, which defaults to you specific pool:

```
from airflow.providers.postgres.operators.postgres import PostgresOperator

class PooledPostgresOperator(PostgresOperator):
    def __init__(self, pool="PostgresOperatorPool", *args, **kwargs):
        super().__init__(pool=pool, *args, **kwargs)

If you don't define any pools in your operator init, it will default to pool `default`

Definitely [check it out](https://airflow.apache.org/docs/apache-airflow/stable/administration-and-deployment/pools.html)!
